crates/nu-errors/src/lib.rs

```diff
@@ -893,27 +893,21 @@ macro_rules! ranged_int {
 
         impl CoerceInto<$ty> for nu_source::Tagged<BigInt> {
             fn coerce_into(self, operation: impl Into<String>) -> Result<$ty, ShellError> {
-                match self.$op() {
-                    Some(v) => Ok(v),
-                    None => Err(ShellError::range_error(
-                        $ty::to_expected_range(),
-                        &self.item.spanned(self.tag.span),
-                        operation.into(),
-                    )),
-                }
+                self.$op().ok_or(ShellError::range_error(
```

Comment:
I'd rather use `ok_or_else` everywhere in this file, otherwise the `ShellError` will always be constructed, even when it's unnecessary.

Reply:
Ahh ok, I didn't realize that the `ok_or()` arguments were always constructed.  Good to know.